### PR TITLE
[loki-stack] Change promtail push url to be better generated

### PR DESCRIPTION
**Pull request by CrazyByDefault**
Addresses #1694 which is affecting promtail deployment in all versions of loki-stack deployments.

### CODE REVIEW

**Comment by CrazyByDefault** [comment]
@jrockway @nervo hey, can I get a review here? this is breaking for a while now.

**Comment by jrockway** [comment]
I don't think either of the people that you mentioned can help you.  We are just random people that have had PRs approved.

**Comment by zanhsieh** [comment]
@CrazyByDefault 
Thanks for contribution. Would you mind to bump the version in Chart.yaml please? CI required so to move forwards.

**Comment by CrazyByDefault** [comment]
> @CrazyByDefault Thanks for contribution. Would you mind to bump the version in Chart.yaml please? CI required so to move forwards.

aye, it's done. can you check now? @zanhsieh

**Review by zanhsieh — APPROVED** [review]

File: charts/loki-stack/Chart.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 apiVersion: "v1"
 name: loki-stack
-version: 2.8.2
+version: 2.8.3
```
Comment (zanhsieh) [inline charts/loki-stack/Chart.yaml:3]:
```suggestion
version: 2.8.4
```

**Review by zanhsieh — CHANGES_REQUESTED** [review]

File: charts/loki-stack/values.yaml [hunk 2]
```diff
@@ -27,7 +27,7 @@ promtail:
     logLevel: info
     serverPort: 3101
     clients:
-      - url: http://{{ .Release.Name }}:3100/loki/api/v1/push
+      - url: http://{{(include "loki.serviceName" .)}}:{{ .Values.loki.service.port }}/loki/api/v1/push
```
Comment (zanhsieh) [inline charts/loki-stack/values.yaml:30]: CI complains:
```
==> Linting loki-stack
[ERROR] templates/: template: loki-stack/charts/promtail/templates/secret.yaml:10:8: executing "loki-stack/charts/promtail/templates/secret.yaml" at <tpl .Values.config.file .>: error calling tpl: error during tpl function execution for "server:\n  log_level: {{ .Values.config.logLevel }}\n  http_listen_port: {{ .Values.config.serverPort }}\n  {{- tpl .Values.config.snippets.extraServerConfigs . | nindent 2 }}\n\nclients:\n  {{- tpl (toYaml .Values.config.clients) . | nindent 2 }}\n\npositions:\n  filename: /run/promtail/positions.yaml\n\nscrape_configs:\n  {{- tpl .Values.config.snippets.scrapeConfigs . | nindent 2 }}\n  {{- tpl .Values.config.snippets.extraScrapeConfigs . | nindent 2 }}\n\nlimits_config:\n  {{- tpl .Values.config.snippets.extraLimitsConfig . | nindent 2 }}\n": template: loki-stack/charts/promtail/templates/secret.yaml:7:6: executing "loki-stack/charts/promtail/templates/secret.yaml" at <tpl (toYaml .Values.config.clients) .>: error calling tpl: error during tpl function execution for "- url: http://{{ .Release.Name }}:{{ .Values.loki.service.port }}/loki/api/v1/push": template: loki-stack/charts/promtail/templates/secret.yaml:1:44: executing "loki-stack/charts/promtail/templates/secret.yaml" at <.Values.loki.service.port>: nil pointer evaluating interface {}.service
```

**Review by zanhsieh — CHANGES_REQUESTED** [review]

**Comment by CrazyByDefault** [comment]
I have pushed what I'm certain is the right way of passing the values down to the promtail subchart, but there seems to be an active bug in Helm that will cause helm and the CI to fail - https://github.com/helm/helm/issues/11168

Somehow, when attempting to execute tpl on [line 398 here](https://github.com/grafana/helm-charts/blob/main/charts/promtail/values.yaml#L398), helm trips up on evaluating `.Release.Name` - 

```
==> Linting charts/loki-stack/
[ERROR] templates/: template: loki-stack/charts/promtail/templates/secret.yaml:10:8: executing "loki-stack/charts/promtail/templates/secret.yaml" at <tpl .Values.config.file .>: error calling tpl: error during tpl function execution for "server:\n  log_level: {{ .Values.config.logLevel }}\n  http_listen_port: {{ .Values.config.serverPort }}\n  {{- tpl .Values.config.snippets.extraServerConfigs . | nindent 2 }}\n\nclients:\n  {{- tpl (toYaml .Values.config.clients) . | nindent 2 }}\n\npositions:\n  filename: /run/promtail/positions.yaml\n\nscrape_configs:\n  {{- tpl .Values.config.snippets.scrapeConfigs . | nindent 2 }}\n  {{- tpl .Values.config.snippets.extraScrapeConfigs . | nindent 2 }}\n\nlimits_config:\n  {{- tpl .Values.config.snippets.extraLimitsConfig . | nindent 2 }}\n": template: loki-stack/charts/promtail/templates/secret.yaml:7:6: executing "loki-stack/charts/promtail/templates/secret.yaml" at <tpl (toYaml .Values.config.clients) .>: error calling tpl: error during tpl function execution for "- url: http://{{ include \"loki.serviceName\" $.Subcharts.loki }}:{{ include \"loki.service.port\"\n    $.Subcharts.loki }}/loki/api/v1/push": template: loki-stack/charts/promtail/templates/secret.yaml:1:17: executing "loki-stack/charts/promtail/templates/secret.yaml" at <include "loki.serviceName" $.Subcharts.loki>: error calling include: template: loki-stack/templates/_helpers.tpl:52:30: executing "loki.serviceName" at <.Release.Name>: invalid value; expected string

Error: 1 chart(s) linted, 1 chart(s) failed
```

**Comment by zanhsieh** [comment]
@CrazyByDefault 
CI still reported error. See below `CI / call-lint-test / lint-test (pull_request) Failing after 20s` for details.

File: charts/loki-stack/values.yaml [hunk 3]
```diff
@@ -20,14 +20,14 @@ loki:
...
-
 promtail:
   enabled: true
   config:
     logLevel: info
     serverPort: 3101
     clients:
-      - url: http://{{ .Release.Name }}:3100/loki/api/v1/push
+      - url: http://{{ .Release.Name }}-loki.{{ .Release.Namespace }}:{{ .Values.loki_svc_port }}/loki/api/v1/push
+      # - url: http://{{ include "loki.serviceName" .Subcharts.loki }}:{{ include "loki.service.port" .Subcharts.loki }}/loki/api/v1/push
```
Comment (nervo) [inline charts/loki-stack/values.yaml:30]: imho, you should remove this comment :)

Comment (CrazyByDefault) [reply charts/loki-stack/values.yaml]: @nervo I don't see it in the latest diff, can you take a look again at the latest diff?

Comment (nervo) [reply charts/loki-stack/values.yaml]: @CrazyByDefault i guess we have access to the same diff :) https://github.com/grafana/helm-charts/pull/1842/files (and no, i don't see it either)

**Pull request closed without merging**